src/main/java/graphql/execution/DataFetcherResult.java

```diff
@@ -0,0 +1,43 @@
+package graphql.execution;
+
+import graphql.GraphQLError;
+import graphql.PublicApi;
+import graphql.schema.DataFetcher;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+
+/**
+ * An object that can be returned from a {@link DataFetcher} that contains both data and errors to be relativized and
+ * added to the final result. This is a useful when your ``DataFetcher`` retrieves data from multiple sources
+ * or from another GraphQL resource.
+ *
+ * @param <T> The type of the data fetched
+ */
+@PublicApi
+public class DataFetcherResult<T> {
+
+    private final T data;
+    private final List<GraphQLError> errors;
+
+    public DataFetcherResult(T data, List<GraphQLError> errors) {
```

Comment:
Assuming `data` can be null? Is there an annotation used by the project to mark such nullable parameters?

Reply:
not that I could find

Reply:
no we don't use/have such an annotation